src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java

```diff
@@ -23,8 +25,25 @@ public ZipStatusController(ZipFileStatusService service) {
     }
     // endregion
 
-    @GetMapping
-    public ZipFileStatus findByFileName(@RequestParam("name") String fileName) {
-        return service.getStatusFor(fileName);
+    @RequestMapping(params = "name")
+    public ResponseEntity<ZipFileStatus> findByFileName(
+        @RequestParam(required = true, value = "name") String fileName
+    ) {
+        if (fileName != null) {
+            return ResponseEntity.ok().body(service.getStatusFor(fileName));
+        }
+        String errorMessage = "'name' can not be empty or is required!";
```

Comment:
not used . 
And as RequestParam is marked as required if the param is empty or null request would not come here.

Reply:
ok. I modify it.

Reply:
if spring handles bad request you do not need 'ResponseEntity'